py3status/modules/sysdata.py

```diff
@@ -81,25 +88,15 @@ def cpu(self):
         # return the cpu total&idle time
         return total_cpu_time, cpu_idle_time
 
-    def memory(self, unit='GiB'):
+    def calc_mem_info(self, unit='GiB', total_mem_kib=float('nan'),
+                      mem_free=float('nan')):
```

Comment:
For this could we not pass the `unit`, `memi` dict, and a list of keys  eg

`calc_mem_info('GiB', memi, ['SwapTotal:', 'SwapFree'])` and just do the conversions in place?